### PR TITLE
SALTO-4899: Fetch Sla

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I added support to fetch SLA

---

None

---
_Release Notes_: 
_Jira Adapter:_
Now support fetch of SLA. 

---
_User Notifications_: 
_Jira Adapter:_
For those who enable JSM, SLA nacls will be visible upon your next fetch

### CODE REVIEW

File: packages/jira-adapter/src/filters/jsm_paths.ts [hunk 1]
```diff
@@ -17,13 +17,14 @@
...
 import { FilterCreator } from '../filter'
-import { CALENDAR_TYPE, QUEUE_TYPE, PORTAL_GROUP_TYPE, REQUEST_TYPE_NAME } from '../constants'
+import { CALENDAR_TYPE, QUEUE_TYPE, PORTAL_GROUP_TYPE, REQUEST_TYPE_NAME, SLA_TYPE_NAME } from '../constants'
 
 const JSM_ELEMENT_DIRECTORY: Record<string, string> = {
   [QUEUE_TYPE]: 'queues',
   [CALENDAR_TYPE]: 'calendars',
   [REQUEST_TYPE_NAME]: 'requestTypes',
   [PORTAL_GROUP_TYPE]: 'portalGroups',
+  [SLA_TYPE_NAME]: 'slas',
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/jsm_paths.ts:27]: maybe `SLAs`?

**Review by alonstern — COMMENTED** [review]
🔥 
There is a `jqlQuery` field under `goals` we probably should parse references from it

File: packages/jira-adapter/src/filters/jsm_paths.ts [hunk 2]
```diff
@@ -17,13 +17,14 @@
...
 import { FilterCreator } from '../filter'
-import { CALENDAR_TYPE, QUEUE_TYPE, PORTAL_GROUP_TYPE, REQUEST_TYPE_NAME } from '../constants'
+import { CALENDAR_TYPE, QUEUE_TYPE, PORTAL_GROUP_TYPE, REQUEST_TYPE_NAME, SLA_TYPE_NAME } from '../constants'
 
 const JSM_ELEMENT_DIRECTORY: Record<string, string> = {
   [QUEUE_TYPE]: 'queues',
   [CALENDAR_TYPE]: 'calendars',
   [REQUEST_TYPE_NAME]: 'requestTypes',
   [PORTAL_GROUP_TYPE]: 'portalGroups',
+  [SLA_TYPE_NAME]: 'SLAs',
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/filters/jsm_paths.ts:27]: If we change path name to `SLAs` maybe it's better to be consistent with typename as well and change to SLA? 
WDYT?

File: packages/jira-adapter/src/filters/jql/jql_references.ts [hunk 3]
```diff
@@ -84,10 +88,34 @@ const getRelativePathJqls = (instance: InstanceElement, pathMap: Record<string,
   return jqlPaths
 }
 
+const getRelativePathJqlsJsm = (instance: InstanceElement, pathMap: Record<string, string[]>): JqlDetails[] => {
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/filters/jql/jql_references.ts:91]: Any chance we can consolidate with `getRelativePathJqls` ?

File: packages/jira-adapter/src/filters/jql/jql_references.ts [hunk 4]
```diff
@@ -84,10 +88,34 @@ const getRelativePathJqls = (instance: InstanceElement, pathMap: Record<string,
...
+  })
+  return jqlPaths
+}
+
 const getJqls = (instance: InstanceElement): JqlDetails[] => {
   if (instanceTypeToMap.has(instance.elemID.typeName)) {
     return getRelativePathJqls(instance, instanceTypeToMap.get(instance.elemID.typeName) as Record<string, string[][]>)
   }
+  if (Object.keys(instanceTypeToString).includes(instance.elemID.typeName)) {
+    return getRelativePathJqlsJsm(instance, instanceTypeToString)
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/filters/jql/jql_references.ts:117]: Anyway I think we should add an example to UT with the new type (as we using a slightly different logic here)

**Review by shir-reifenberg — COMMENTED** [review]
Looks good 😎 , added a few comments. 
Might be out of context, but can we add service URL? and do we support alias in JSM (and should add to this type) or not yet?

File: packages/jira-adapter/src/filters/jsm_paths.ts (hunk 2 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/jsm_paths.ts:27]: Agreed

File: packages/jira-adapter/src/filters/jql/jql_references.ts (hunk 3 above)
Comment (shir-reifenberg) [reply packages/jira-adapter/src/filters/jql/jql_references.ts:91]: Talked IRL

**Review by shir-reifenberg — APPROVED** [review]
💯

**Pull request merged**